internal/merkle/clock/clock.go

```diff
@@ -86,20 +87,20 @@ func (mc *MerkleClock) AddDelta(
 	delta.SetPriority(height)
 	block := coreblock.New(delta, links, heads...)
 
-	isEncrypted, err := mc.checkIfBlockEncryptionEnabled(ctx, block.Delta.GetFieldName(), heads)
+	fieldName := immutable.None[string]()
+	if block.Delta.GetFieldName() != "" {
+		fieldName = immutable.Some(block.Delta.GetFieldName())
+	}
+	blockEnc, err := mc.determineBlockEncryptionData(ctx, string(block.Delta.GetDocID()), fieldName, heads)
```

Comment:
todo: This is the issue referenced above regarding the `generatedKeys` slice in the `encryptor` struct.

The `generatedKeys` is used to keep track of the locally generated keys to quickly get them for later use. However, this isn't necessary because we already have the `heads` here at this call site, which means we can get the last keyID of the head (or deterministically choose a key from multiple heads if they exist, eg: lexigraphical CID compare).